File: update_icons.py
import os
from PIL import Image

# Source image path
source_path = "/home/mimura/.gemini/antigravity/brain/d2bbc4c0-12fc-42bc-ab01-c9b0d7380029/igopon_text_icon_1764282064857.png"
public_dir = "/home/mimura/projects/igopon/public"

# Target sizes and filenames
targets = [
    ("pwa-192x192.png", (192, 192)),
    ("pwa-512x512.png", (512, 512)),
    ("maskable-icon-512x512.png", (512, 512)),
    ("favicon.png", (64, 64)),
    ("apple-touch-icon.png", (180, 180))
]

def process_images():
    if not os.path.exists(source_path):
        print(f"Error: Source image not found at {source_path}")
        return

    try:
        with Image.open(source_path) as img:
            print(f"Opened source image: {source_path}")
            
            for filename, size in targets:
                target_path = os.path.join(public_dir, filename)
                # Resize using LANCZOS for high quality downsampling
                resized_img = img.resize(size, Image.Resampling.LANCZOS)
                resized_img.save(target_path, "PNG")
                print(f"Saved {filename} ({size[0]}x{size[1]}) to {target_path}")
                
    except ImportError:
        print("Error: PIL (Pillow) is not installed. Please install it with 'pip install Pillow'.")
    except Exception as e:
        print(f"An error occurred: {e}")

if __name__ == "__main__":
    process_images()

File: update_icons.js
import { Jimp } from 'jimp';
import path from 'path';
import fs from 'fs';

// Use the existing high-res icon as source since we want to modify it
const sourcePath = "/home/mimura/projects/igopon/public/maskable-icon-512x512.png";
const publicDir = "/home/mimura/projects/igopon/public";

const ZOOM_FACTOR = 1.5; // Adjust this to control how much we zoom in

const targets = [
    { name: "pwa-192x192.png", width: 192, height: 192 },
    { name: "pwa-512x512.png", width: 512, height: 512 },
    { name: "maskable-icon-512x512.png", width: 512, height: 512 },
    { name: "favicon.png", width: 64, height: 64 },
    { name: "apple-touch-icon.png", width: 180, height: 180 }
];

async function processImages() {
    try {
        if (!fs.existsSync(sourcePath)) {
            console.error(`Source image not found at ${sourcePath}`);
            return;
        }

        const image = await Jimp.read(sourcePath);
        console.log(`Opened source image: ${sourcePath}`);

        for (const target of targets) {
            const targetPath = path.join(publicDir, target.name);
            const resized = image.clone().resize({ w: target.width, h: target.height });
            await resized.write(targetPath);
            console.log(`Saved ${target.name} (${target.width}x${target.height}) to ${targetPath}`);
        }
    } catch (error) {
        console.error("An error occurred:", error);
    }
}

processImages();
